axelor-message/src/main/java/com/axelor/apps/message/service/TemplateService.java

```diff
@@ -97,4 +111,46 @@ public String processContent(
     maker.setContext(bean, context, beanName);
     return maker.make();
   }
+
+  public void addItemToReferenceSelection(MetaModel model) {
+    MetaSelect metaSelect =
+        Beans.get(MetaSelectRepository.class)
+            .findByName("print.template.line.test.reference.select");
+    List<MetaSelectItem> items = metaSelect.getItems();
+    if (items != null && !items.stream().anyMatch(x -> x.getValue().equals(model.getFullName()))) {
+      MetaSelectItem metaSelectItem = new MetaSelectItem();
+      metaSelectItem.setTitle(model.getName());
+      metaSelectItem.setValue(model.getFullName());
+      metaSelectItem.setSelect(metaSelect);
+      saveMetaSelectItem(metaSelectItem);
+    }
+  }
+
+  @Transactional
+  public void saveMetaSelectItem(MetaSelectItem metaSelectItem) {
+    Beans.get(MetaSelectItemRepository.class).save(metaSelectItem);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Transactional
+  public Message generateDraftMessage(Template template, MetaModel metaModel)
+      throws ClassNotFoundException, AxelorException {
+
+    if (metaModel == null) {
+      return null;
+    }
+    String model = metaModel.getFullName();
+    Model modelObject = null;
+
+    if (StringUtils.notEmpty(model)) {
+      Class<? extends Model> modelClass = (Class<? extends Model>) Class.forName(model);
+      modelObject = JPA.find(modelClass, Long.valueOf(template.getReferenceId().toString()));
+    }
+    try {
+      return Beans.get(TemplateMessageService.class).generateMessage(modelObject, template);
```

Comment:
use >> Message generateMessage(Model model, Template template, Boolean isTemporaryMessage)
this will generate unsaved message (i.e. without id)